purchases/src/main/kotlin/com/revenuecat/purchases/customercenter/ScreenMapSerializer.kt

```diff
@@ -0,0 +1,37 @@
+package com.revenuecat.purchases.customercenter
+
+import com.revenuecat.purchases.common.warnLog
+import com.revenuecat.purchases.customercenter.CustomerCenterConfigData.Screen
+import com.revenuecat.purchases.customercenter.CustomerCenterConfigData.Screen.ScreenType
+import kotlinx.serialization.KSerializer
+import kotlinx.serialization.builtins.MapSerializer
+import kotlinx.serialization.descriptors.SerialDescriptor
+import kotlinx.serialization.encoding.Decoder
+import kotlinx.serialization.encoding.Encoder
+import kotlinx.serialization.json.JsonDecoder
+import kotlinx.serialization.json.jsonObject
+
+internal object ScreenMapSerializer : KSerializer<Map<ScreenType, Screen>> {
```

Comment:
Had to write a custom deserializer here for the map of screen types to screen, to make sure if we start receiving unknown screen types in the future, we still can parse the known ones.

Reply:
Great foresight! Does the builtin deserializer fail even with `ignoreUnknownKeys`?

Reply:
Right, if there was a new screen, like `TEST`, without this code it would fail with `com.revenuecat.purchases.customercenter.CustomerCenterConfigData.Screen.ScreenType does not contain element with name 'TEST' at path $.customer_center.screens`

Reply:
Oh right, because of the enum. Makes sense!